app/Controllers/userController.php

```diff
@@ -103,8 +103,9 @@ public static function createUser($new_user_name, $passwordPlain, $apiPasswordPl
 		if (!is_array($userConfig)) {
 			$userConfig = array();
 		}
+        $aValid = array('-', '_', '.');
 
-		$ok = ($new_user_name != '') && ctype_alnum($new_user_name);
+		$ok = ($new_user_name != '') && ctype_alnum(str_replace($aValid, '', $new_user_name));
```

Comment:
I think we should extract that as a new function to validate usernames, since the same code is needed at several places.